control/pid.go

```diff
@@ -22,25 +22,51 @@ func (l *Loop) newPID(config BlockConfig, logger logging.Logger) (Block, error)
 
 // BasicPID is the standard implementation of a PID controller.
 type basicPID struct {
-	mu       sync.Mutex
-	cfg      BlockConfig
-	error    float64
-	kI       float64
-	kD       float64
-	kP       float64
-	int      float64
+	mu     sync.Mutex
+	cfg    BlockConfig
+	logger logging.Logger
+	// used by the single input/output controller
+	error float64
+	kI    float64
+	kD    float64
+	kP    float64
+	int   float64
+	tuner pidTuner
+
+	// MIMO gains + state
+	PIDSets  []*PIDConfig
+	tuners   []*pidTuner
+	useMulti bool
+
+	// used by both
 	y        []*Signal
 	satLimUp float64 `default:"255.0"`
 	limUp    float64 `default:"255.0"`
 	satLimLo float64
 	limLo    float64
-	tuner    pidTuner
-	tuning   bool
-	logger   logging.Logger
 }
 
+// GetTuning returns whether the PID block is currently tuning any signals.
 func (p *basicPID) GetTuning() bool {
-	return p.tuning
+	// using locks so we do not check for tuning mid reconfigure or mid tune
```

Comment:
is this comment actually what's happening? currently `setup_control.go` calls `MonitorTuning()`, which calls `loop.GetTuning()`, which calls this pid block `GetTuning()` function. I don't actually think the locks will cause an issue here, but this function _is_ getting called mid tune. maybe just reword comment because I don't think these locks are actually stopping this function if the loop is actively tuning

Reply:
the locks are in place so we do not check the value of `tuning` while we are doing a `pidTunerStep`

if we didn't have a lock, then we would be able to read the value of `tuning` while it was being modified, which would result in a race

Reply:
yeah that makes sense. I guess I just interpreted "mid tune" as the entire tuning process. maybe update the comment to clarify you mean mid tuning step. especially when it's grouped with configure it seems like the whole picture of tuning a component